### PR TITLE
Scroll back up on Roadmap causes blinking step icon

### PR DESCRIPTION
**Pull request by justtry**
https://github.com/x-team/auto/issues/818

Fixing scroll logic and preventing blinking

### CODE REVIEW

File: src/components/ui/RoadmapTimelineElement.js [hunk 1]
```diff
@@ -71,20 +71,21 @@ const animatedStyles = {
...
+      transition: box-shadow ease-in-out .2s
     }
-
-    @keyframes fill {
-      100% {
-        box-shadow: inset 0 0 0 30px ${theme.baseRed}
-      }
+  `),
+  checkmarkActive: cmz(`
+    & {
```
Comment (nicksp) [inline src/components/ui/RoadmapTimelineElement.js]: If changes are just affect the element itself, for a shorther form, `& {}` notation can be skipped, so the final css looks like this:

```
checkmarkActive: cmz('box-shadow: ....')
```

File: src/utils/helpers.js [hunk 2]
```diff
@@ -11,15 +11,14 @@ export function throttle (callback, timeout) {
...
   // Viewport offset
   const scrollPosition = window.scrollY
   const docViewBottom = scrollPosition + window.innerHeight
 
   // Element's position relative to the document
-  const elemTop = scrollPosition + top
   const elemBottom = scrollPosition + bottom
 
-  return direction < 0 ? elemTop < docViewBottom : elemBottom <= docViewBottom
+  return elemBottom <= docViewBottom
```
Comment (nicksp) [inline src/utils/helpers.js]: right, so you got fixed it by simplifying the condition :) That's what I had in mind when first developed it. Just wanted to have different behaviour based on scroll direction. But apparently, it was buggy :-)

If you do change this way, please update a bit further. You now no longer depend on a scroll direction (`direction` prop), so you'll also need to update a calling `Roadmap` module to remove unneeded code and logic around the calculation of scroll direction, as it's not needed anymore.

Also, whenever you update `auto-ui`, make sure to run a live version (`localhost:6060`) in a browser and update failing tests, so those get committed as well 😄

File: src/components/ui/RoadmapTimelineElement.js [hunk 3]
```diff
@@ -99,8 +100,8 @@ class RoadmapTimelineElement extends PureComponent<Props> {
     const cx = {
       active: isDone ? animatedStyles.timelineActive : '',
       box: animatedStyles.checkmarkBox,
-      checkmark: isDone ? animatedStyles.checkmark : '',
-      checkmarkCheck: isDone ? animatedStyles.checkmarkCheck : '',
+      checkmark: `${animatedStyles.checkmark} ${isDone ? animatedStyles.checkmarkActive : ''}`,
+      checkmarkCheck: `${animatedStyles.checkmarkCheck} ${isDone ? animatedStyles.checkmarkCheckActive : ''}`,
```
Comment (nicksp) [inline src/components/ui/RoadmapTimelineElement.js]: Also, could we please simplify classes used here, seems a bit confusing.
Is that possible to have two classes as before: `checkmark` - for a completed checkmark and `checkmarkCheck` for a completed underlying check icon?

File: src/utils/helpers.js (hunk 2 above)
Comment (justtry) [reply src/utils/helpers.js]: thx, I think in both direction behaviour should be same - that is most natural for viewer

File: src/components/ui/RoadmapTimelineElement.js (hunk 3 above)
Comment (justtry) [reply src/components/ui/RoadmapTimelineElement.js]: hey @nicksp, I think it's good to have default styles for such elements in default state (checkmark(for the circle) and checkmarkCheck(not checked state - just for check mark inside)) and for active state (checkmarkActive and checkmarkCheckActive), so it's more clear for styles editing

Comment (nicksp) [reply src/components/ui/RoadmapTimelineElement.js]: Sounds reasonable Julia :-) Let's keep it the way you proposed then 👍

**Comment by nicksp** [comment]
is it ready for re-review ;-) ?

File: src/components/ui/RoadmapTimelineElement.js (hunk 1 above)
Comment (justtry) [reply src/components/ui/RoadmapTimelineElement.js]: Updated, thx

**Comment by justtry** [comment]
Yes, thanks^ ready for re-review

File: src/components/ui/Roadmap.js [hunk 4]
```diff
@@ -157,15 +157,10 @@ class Roadmap extends PureComponent<Props, State> {
     window.removeEventListener('scroll', this.throttledScrollHandler)
   }
 
-  detectScrollDirection = () => {
-    // 1 — scroll down; -1 — scroll up
-    return this.lastScrollPos > window.scrollY ? -1 : 1
```
Comment (nicksp) [inline src/components/ui/Roadmap.js]: Don't forget to remove this variable `lastScrollPos` also, as it was used for the purpose of detecting scroll direction.

Comment (justtry) [reply src/components/ui/Roadmap.js]: Haven't noticed those tests at all :sweat_smile: now totally will pay attention to it, thx for screen @nicksp !

**Comment by justtry** [comment]
Updated, ready for recheck.

File: src/components/ui/Roadmap.js (hunk 4 above)
Comment (nicksp) [reply src/components/ui/Roadmap.js]: no worries 😃

**Pull request merged**